cmd/utils.go

```diff
@@ -318,3 +325,89 @@ func isURLContains(srcURL, tgtURL, sep string) bool {
 	}
 	return false
 }
+
+// Return the file system attribute as string; containing mode,
+// uid, gid, uname, Gname, atime, mtime, ctime and md5
+func getFileSystemAttrs(file string) (string, error) {
+	var uname, gname, atime, mtime, ctime, mode, uid, gid string
+	fi, err := os.Stat(file)
+	if err != nil {
+		return "", err
+	}
+	st := fi.Sys().(*syscall.Stat_t)
+	mode = fmt.Sprint(st.Mode)
+	uid = fmt.Sprint(st.Uid)
+	gid = fmt.Sprint(st.Gid)
+
+	i, err := user.LookupId(strconv.FormatUint(uint64(st.Uid), 10))
+	if err != nil {
+		return "", err
+	}
+	uname = i.Username
+
+	g, err := user.LookupGroupId(strconv.FormatUint(uint64(st.Gid), 10))
+	if err != nil {
+		return "", err
+	}
+	gname = g.Name
+
+	t, err := times.Stat(file)
+	if err != nil {
+		return "", err
+	}
+	atime = fmt.Sprint(t.AccessTime().Unix())
+	mtime = fmt.Sprint(t.ModTime().Unix())
+
+	if t.HasChangeTime() {
+		ctime = fmt.Sprint(t.ChangeTime().Unix())
+	}
+
+	md5, err := hashFilemd5(file)
+	if err != nil {
+		return "", err
+	}
+	fileAttr := fmt.Sprintf("atime:%s/ctime:%s/gid:%s/gname:%s/md5:%s/mode:%s/mtime:%s/uid:%s/uname:%s", atime, ctime, gid, gname, md5, mode, mtime, uid, uname)
+
+	return fileAttr, nil
+}
+
+// Return the md5 of the file as string
+func hashFilemd5(filePath string) (string, error) {
```

Comment:
This will slow down the process, we don't need to save md5sum of the objects we can totally ignore this part and not implement it.